sources/api/schnauzer/src/helpers.rs

```diff
@@ -90,6 +92,44 @@ lazy_static! {
 const PAUSE_FALLBACK_REGISTRY: &str = "602401143452";
 const PAUSE_FALLBACK_REGION: &str = "us-east-1";
 
+lazy_static! {
+    /// A map to tell us which partition from for a given region.
+    static ref PARTITION_MAP: HashMap<&'static str, &'static str> = {
+        let mut m = HashMap::new();
```

Comment:
Non-blocking comment: I was looking at `aws-sdk-rust` to see if we can get this information from their region types but it seems like the answer is no. So I guess we'll have to live with this. I really don't like the proliferation of these kinds of static hash maps that should be sourcing the truth from elsewhere.

Reply:
I agree. I did add a method of fetching the partition from IMDS in https://github.com/bottlerocket-os/bottlerocket/pull/2224, but based on prior discussions the consensus was this might be more performant, especially if multiple settings need partitional context.

Reply:
Another non-blocking comment: I would also prefer not to see a large hashmap like this, and would prefer that it only list the exceptions (in this case, the aws-cn and aws-us-gov regions) rather than need to be updated each time a new region was built.